src/occa/internal/modes/cuda/device.cpp

```diff
@@ -59,24 +59,11 @@ namespace occa {
       kernelProps["compiler"] = compiler;
       kernelProps["compiler_flags"] = compilerFlags;
 
-#if CUDA_VERSION < 5000
-      OCCA_CUDA_ERROR("Device: Getting CUDA device arch",
-                      cuDeviceComputeCapability(&archMajorVersion,
-                                                &archMinorVersion,
-                                                cuDevice));
-#else
-      OCCA_CUDA_ERROR("Device: Getting CUDA device major version",
-                      cuDeviceGetAttribute(&archMajorVersion,
-                                           CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR,
-                                           cuDevice));
-      OCCA_CUDA_ERROR("Device: Getting CUDA device minor version",
-                      cuDeviceGetAttribute(&archMinorVersion,
-                                           CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MINOR,
-                                           cuDevice));
-#endif
-
+      getDeviceArchVersion(cuDevice, archMajorVersion, archMinorVersion);
       archMajorVersion = kernelProps.get("arch/major", archMajorVersion);
       archMinorVersion = kernelProps.get("arch/minor", archMinorVersion);
+
+      arch = getDeviceArch(cuDevice);
```

Comment:
Oh, thats a very good point. Thanks for catching that. I'll fix this up and make sure settings for architecture are used

Reply:
awesome, thanks!

Reply:
Does it make sense to determine the target architecture on a per-kernel basis, rather than in the device constructor? Currently the target arch flags are essentially determined in the device constructor, but is it reasonable for a user to pass a target arch for a singular kernel build?

Reply:
I think that makes sense - to handle arch that the hardware supports and the arch that a user wants to build for, separately.  Perhaps useful when a user has kernels to build for different target architectures (as each kernel is optimized for a specific arch, perhaps rare in real app. though?). If a user doesn't specify the target arch for a kernel, then it makes sense to use a default - the highest one that current hardware supports from the device constructor, IMO.